uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/BrokerHost.java

```diff
@@ -292,16 +327,111 @@ private final String basicButtonHandler(@NonNull final String resourceButtonId,
             UiAutomatorUtils.handleButtonClick("android:id/button1");
         }
     }
-
-    @Override
-    public void setFlights(@Nullable final String flightsJson) {
-        Logger.i(TAG, "Set Flights..");
+    /**
+     * Gets all the accounts added
+     */
+    public List<String> getAllAccounts(boolean expectingMultipleAccounts) {
         launch();
-        // Make the set flights UI visible on screen
-        UiAutomatorUtils.obtainChildInScrollable("Set Flights");
-        // input flights string in flights input box
-        UiAutomatorUtils.handleInput("com.microsoft.identity.testuserapp:id/editTextFlights", flightsJson);
-        // Click Set Flights button
-        UiAutomatorUtils.handleButtonClick("com.microsoft.identity.testuserapp:id/setFlightsButton");
+
+        final String resourceButtonId = "com.microsoft.identity.testuserapp:id/buttonGetAccounts";
+        final String textId = "AccountName";
+        if (shouldHandleFirstRun) {
+            handleFirstRun(); // handle first run experience
+        }
+        UiAutomatorUtils.handleButtonClick(resourceButtonId);
+        // Look for the dialog box
+        final UiObject dialogBox = UiAutomatorUtils.obtainUiObjectWithResourceId(
+                "android:id/message"
+        );
+        Assert.assertTrue(dialogBox.exists());
+
+        // As of now we are testing only for 2 accounts.
+        List<String> accounts = new ArrayList<>();
+        final String accountName = getDialogBoxText(dialogBox, textId);
+        if (accountName != null) {
+            accounts.add(accountName);
+        }
+        if (expectingMultipleAccounts) {
+            final String accountName2 = getDialogBoxText(dialogBox, textId);
+            if (accountName2 != null)
+                accounts.add(accountName2);
+        }
+        return accounts;
+    }
+
+    /**
+     * Removes the added account
+     */
+    public void removeAccount(String username) {
+        final UiObject removeAccount = UiAutomatorUtils.obtainUiObjectWithResourceId(
+                "com.microsoft.identity.testuserapp:id/buttonRemoveAccount"
+        );
+        UiAutomatorUtils.obtainChildInScrollable("someone@contoso.com");
```

Comment:
Is this intentional?

Reply:
do you mean searching for the input box with hint as "someone@contoso.com"?

Reply:
Yes. I'm not sure what we're trying to do here. It seems we're looking for some UiObject and then totally forgetting about it.

Reply:
this was done to first scroll and then fill with resourceId. Changed it to reuse the UIObject obtained after scrolling.